developer-tools/terraform/quickstart.mdx

```diff
@@ -0,0 +1,118 @@
+dates:
+  validation: 2023-10-24
+  posted: 2023-10-24
+categories:
+  - devtools
+---
+
+[Terraform](https://www.terraform.io/) is an Infrastructure as Code (IAC) software environment tool that allows you to easily build and maintain your cloud resources. You can use it to safely and efficiently deploy Scaleway resources in a declarative way.
+
+Check out the [Scaleway Terraform registry](https://registry.terraform.io/providers/scaleway/scaleway/latest/docs) documentation for more information.
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [made your first request using the Scaleway API](https://www.scaleway.com/en/developers/api/#quickstart:-first-request)
+  - You have [installed Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli#install-terraform)
+</Message>
+
+## Creating and Deploying Scaleway resources using Terraform
+
+For the sake of this example, we are going to create a Kubernetes Kapsule cluster 
+
+### Creating the Terraform configuration file
+
+1. Create a new file called `main.tf` in a new folder. 
+
+2. Open the file in a code editor and paste the following block to define Scaleway as the provider for this deployment:
+    ```
+    terraform {
+      required_providers {
+        scaleway = {
+          source = "scaleway/scaleway"
+        }
+      }
+      required_version = ">= 0.13"
+    }
+    ```
+3. Add the following block and replace the placeholder with your project ID to determine in which project the resources are created:
+    ```
+    variable "project_id" {
+      type        = string
+      description = "<PROJECT_ID>"
+    }
+    ```
+4. Add the following blocks to create a Private Network, a Kubernetes cluster and its associated pool:
+    ```
+    resource "scaleway_vpc_private_network" "my-pn" {
+      name = "my-private-network"
+    }
+
+    resource "scaleway_k8s_cluster" "my-cluster" {
+      name    = "my-cluster"
+      type    = "kapsule"
+      version = "1.28.2"
+      cni     = "cilium"
+      private_network_id = scaleway_vpc_private_network.my-pn.id
+      delete_additional_resources = false
+    }
+
+    resource "scaleway_k8s_pool" "my-pool" {
+      cluster_id  = scaleway_k8s_cluster.my-cluster.id
+      name        = "my-pool"
+      node_type   = "DEV1-M"
+      size        = 1
+      min_size    = 0
+      max_size    = 1
+      autoscaling = true
+      autohealing = true
+    }
+    ```
+    <Message type="note">
+    Refer to the [Scaleway provider documentation on the Terraform Registry](https://registry.terraform.io/providers/scaleway/scaleway/latest/docs) to learn how to deploy other resources.
+    </Message>
+5. Save your changes.
+
+Your configuration is now ready to be deployed on Scaleway's infrastructure.
+
+### Deploying your configuration using Terraform
+
+1. Run the following command to initialize Terraform in the folder:
```

Comment:
This can be misleading as it implies the folder to access is a child of the current folder (which is not necessarily the case) + it's developer doc so I'm not sure it's worth adding the `cd` part :)

Reply:
(But I'll add the first part)